kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/authorization/ClientAuthzIT.java

```diff
@@ -0,0 +1,935 @@
+        }
+
+        public Outcome<Void> seekToBeginning(List<TopicPartition> partitions) {
+            Outcome<Void> outcome;
+            try {
+                consumer.seekToBeginning(partitions);
+                outcome = new Success<>(null);
+            }
+            catch (RuntimeException e) {
+                outcome = Fail.of(e);
+            }
+            return trace("seekToBeginning", outcome);
+        }
+
+        public Outcome<ConsumerRecords<K, V>> poll() {
+            Outcome<ConsumerRecords<K, V>> outcome;
+            try {
+                Duration timeout = Duration.ofSeconds(3);
+                ConsumerRecords<K, V> poll = consumer.poll(timeout);
+
+                outcome = new Success(poll);
+            }
+            catch (KafkaException e) {
+                outcome = Fail.of(e);
+            }
+            trace("poll", outcome.map(PrettyRecords::new));
+            return outcome;
+        }
+
+        public Outcome<Void> commitSync() {
+            Outcome<Void> outcome;
+            try {
+                // send to topicA
+                consumer.commitSync();
+                outcome = new Success<>(null);
+            }
+            catch (KafkaException e) {
+                outcome = Fail.of(e);
+            }
+            return trace("commitSync", outcome);
+        }
+
+        public Outcome<Void> subscribe(String topic) {
+            Outcome<Void> outcome;
+            try {
+                // send to topicA
+                consumer.subscribe(List.of(topic));
+                outcome = new Success<>(null);
+            }
+            catch (KafkaException e) {
+                outcome = Fail.of(e);
+            }
+            return trace("subscribe", outcome);
+        }
+
+        public Outcome<ConsumerGroupMetadata> groupMetadata() {
+            Outcome<ConsumerGroupMetadata> outcome;
+            try {
+                outcome = new Success<>(consumer.groupMetadata());
+            }
+            catch (RuntimeException e) {
+                outcome = Fail.of(e);
+            }
+            return trace("groupMetadata", outcome.map(m -> new ConsumerGroupMetadata(m.groupId(), m.generationId(), "CLOBBERED", m.groupInstanceId())));
+        }
+    }
+
+    static Map<String, Object> adding(Map<String, Object> map, Map<String, Object> and) {
+        var result = new HashMap<>(map);
+        result.putAll(and);
+        return result;
+    }
+
+    interface Prog {
+
+        void run(ClientFactory clientFactory) throws InterruptedException;
+    }
+
+    record AdminProg() implements Prog {
+
+        @Override
+        public void run(ClientFactory clientFactory) {
+            String topicA = TOPIC_A;
+            String topicB = TOPIC_B;
+            TopicPartition topicAPartition0 = new TopicPartition(topicA, 0);
+            TopicPartition topicAPartition1 = new TopicPartition(topicA, 1);
+
+            String setupUser = ALICE;
+            String adminUser = ALICE;
```

Comment:
all the progs use `ALICE` for all clients, is there anything using the `EVE` actor? Also, up top we set up some rules for `Bob`, but he doesn't feature in the tests.